controller/monitor/fake_node_monitor.go

```diff
@@ -0,0 +1,174 @@
+package monitor
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"sync"
+	"time"
+
+	"github.com/jinzhu/copier"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/longhorn/longhorn-manager/datastore"
+	longhorn "github.com/longhorn/longhorn-manager/k8s/pkg/apis/longhorn/v1beta2"
+	"github.com/longhorn/longhorn-manager/types"
+	"github.com/longhorn/longhorn-manager/util"
+)
+
+const (
+	FakeNodeMonitorSyncPeriod = 30 * time.Second
+
+	TestDiskID1 = "fsid"
+
+	TestOrphanedReplicaDirectoryName = "test-volume-r-000000000"
+)
+
+type FakeNodeMonitor struct {
+	*baseMonitor
+
+	nodeName string
+
+	collectedDataLock sync.RWMutex
+	collectedData     map[string]*CollectedDiskInfo
+
+	syncCallback func(key string)
+
+	getDiskStatHandler GetDiskStatHandler
+	getDiskConfig      GetDiskConfig
+	generateDiskConfig GenerateDiskConfig
+}
+
+func NewFakeNodeMonitor(logger logrus.FieldLogger, ds *datastore.DataStore, node *longhorn.Node, syncCallback func(key string)) (*FakeNodeMonitor, error) {
```

Comment:
You can input `getDiskStatHandler`, `getDiskConfig`, and `generateDiskConfig`. Then you don't need this Fake Monitor.

Reply:
But the collections of orphaned directory names in the two monitors are different, I have no better idea handle them.

Reply:
Addressed!